table/server/master/src/main/java/alluxio/master/table/UnpartitionedTableScheme.java

```diff
@@ -0,0 +1,63 @@
+/*
+ * The Alluxio Open Foundation licenses this work under the Apache License, version 2.0
+ * (the "License"). You may not use this work except in compliance with the License, which is
+ * available at www.apache.org/licenses/LICENSE-2.0
+ *
+ * This software is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND,
+ * either express or implied, as more fully set forth in the License.
+ *
+ * See the NOTICE file distributed with this work for information regarding copyright ownership.
+ */
+
+package alluxio.master.table;
+
+import alluxio.grpc.table.Storage;
+import alluxio.grpc.table.UdbTableInfo;
+
+import com.google.common.base.Preconditions;
+
+import java.util.Arrays;
+import java.util.List;
+
+/**
+ * Unpartitoned table scheme.
+ */
+public class UnpartitionedTableScheme implements PartitionScheme {
+  private Partition mPartition;
+  private UdbTableInfo mBaseInfo;
+
+  /**
+   * Constructor for UnpartitionedTableScheme.
+   *
+   * @param partitions a list of partitions
+   * @param tableInfo table info
+   */
+  public UnpartitionedTableScheme(List<Partition> partitions, UdbTableInfo tableInfo) {
+    Preconditions.checkArgument(partitions.size() == 1);
+    mPartition = partitions.get(0);
+    mBaseInfo = tableInfo;
+  }
+
+  @Override
+  public List<Partition> getPartitions() {
+    return Arrays.asList(mPartition);
+  }
+
+  @Override
+  public UdbTableInfo getTableLayout() {
+    if (mBaseInfo.hasHiveTableInfo()) {
```

Comment:
im not sure this will work, since the alluxio layer should not have compile-time knowledge of udb types.

Reply:
We have to have a way to update the tableInfo proto though. Anytime we add a udb type, we would need to change the `UdbTableInfo` proto to include an additional type anyways. I am trying to avoid using a similar strategy to layout for tableInfo.